ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorizedRowBatchCtx.java

```diff
@@ -574,10 +574,12 @@ public void addPartitionColsToBatch(ColumnVector col, Object value, int colIndex
         if (value == null) {
           dv.noNulls = false;
           dv.isNull[0] = true;
-          dv.isRepeating = true;
         } else {
-          dv.fill(((HiveDecimal) value).longValue());
+          // fill serialized decimal value
+          dv.set(0, (HiveDecimal) value);
```

Comment:
why do we hardcode with 0?

Reply:
My bad, I missed that. Should we change herhttps://github.com/apache/hive/pull/5953/files#diff-2773a0f7ae920d36ecdaa185be6df7007e45ce9fb184f2b8754dbfc3038c00b1R590e as well?

Reply:
@Indhumathi27 could you please drop the `fill(long value)` from  Decimal64ColumnVector?